### PR TITLE
GFNet Vision Transformer for Alzheimer's Disease Classification - 48007506

### PR DESCRIPTION
**Pull request by MorrisWang9**
Implementation of a Global Filter Network classification model that can label a set of MRI brain scans as either Normal Cognitive or Alzheimer's Disease.

### CODE REVIEW

**Comment by wangzhaomxy** [comment]
**<This is an initial inspection, no action is required at this point.>**

**File Organizing:** Well-organized files.

**Problem Solving:**

- The algorithm can solve the problem, but it's not GFNet. 
- Accuracy in testing dataset: 0.7511. 

**Model and functions:**

- It uses PyTorch to construct the model and functions. However, the model is not GFNet.The constructed model is based on a pre-trained ResNet18 classification model, moving the last MLP layer and then connecting a GF block.
- With data augmentation.

**Code design:** Good.

**Code comment and docstring:**

- Good code comments 
- No function docstrings

**Difficulty:** Hard.

**Additional Comments:**

- Detailed experiment analysis.
- insufficient commits.
- Good ReadMe design, but need a conclusion for the report.

**Comment by aniketgupta17** [comment]
Observational Feedback

Pull Request:
Correctly created the Pull request from Topic Recognition Branch .
The pull request includes NO description about the file structure .
No previous feedback incorporated . 

File Organizing: Well-organized files.

Commit Log:
Commit messages could be more progressive for the Recognition Problem solved using 4 files . 
Commits are not  regularly made and insufficient . 

Documentation:

Code comments and docstrings  are not included.
Proper GitHub markdown formatting is used, with organized headings, lists, and code blocks.
Can add about Conclusion and  future improvements in the ReadMe.

**Comment by gayanku** [comment]
## Marking
<table><tbody>
  <tr><td colspan='3'><b>Good/OK/Fair Practice (Design/Commenting, TF/Torch Usage)</b></td> </tr>
	<tr><td></td><td>Good design and implementation. </td><td></td></tr>
	<tr><td></td><td>Spacing and comments. </td><td></td></tr>
	<tr><td></td><td>Header blocks. </td><td></td></tr>
	
  <tr><td colspan='3'><b>Recognition Problem</b></td></tr>
	<tr><td></td><td>OK solution to problem. Not GFNet</td><td>-3</td></tr>
	<tr><td></td><td>Driver Script present. </td><td></td></tr>
	<tr><td></td><td>File structure present. </td><td></td></tr>
	<tr><td></td><td>Good Usage & Demo & Visualisation & Data usage. </td><td></td></tr>
	<tr><td></td><td>Module present. </td><td></td></tr>
	<tr><td></td><td>Commenting missing. Minimal doc strings.</td><td>-1</td></tr>
	<tr><td></td><td>No Data leakage found. Uses Data Aug</td><td></td></tr>
	<tr><td></td><td>Difficulty : Hard.  GFNet (Hard Difficulty)</td><td></td></tr>
	
  <tr><td colspan='3'><b>Commit Log</b></td></tr>
	<tr><td></td><td>Some/Adequate Meaningful commit messages. </td><td>-1</td></tr>
	<tr><td></td><td>Good Progressive commits. </td><td></td></tr>
	
  <tr><td colspan='3'><b>Documentation</b></td></tr>
	<tr><td></td><td>Readme :Good. </td><td></td></tr>
	<tr><td></td><td>Model/technical explanation :Good. </td><td></td></tr>
	<tr><td></td><td>Description and Comments :Good. </td><td></td></tr>
	<tr><td></td><td>Markdown used and PDF submitted. PDF Checked.</td><td></td></tr>
	
  <tr><td colspan='3'><b>Pull Request</b></td></tr>
	<tr><td></td><td>Successful Pull Request (Working Algorithm Delivered on Time in Correct Branch). </td><td></td></tr>
	<tr><td></td><td>No Feedback required. </td><td></td></tr>
	<tr><td></td><td>Request Description is adequate. </td><td>-1</td></tr>
  <tr><td colspan='2'><b>TOTAL</b></td><td>-6</td></tr>
</tbody></table>

Marked as per the due date and changes after which aren't necessarily allowed to contribute to grade for fairness.
_Subject to approval from Shakes_

**Comment by hanemma7moud** [comment]
Chiao-Yu Wang

**Comment by shakes76** [comment]
Commit by due date did not have a proper GFNet, fixed after marking. This is not eligible for feedback marks, though other things were. Thus not hard difficult -5 nor normal difficulty -5, incorrect model -2, +2 feedback, results not reflecting of solution required. -2
